src/test/java/name/webdizz/sonar/grammar/spellcheck/JavaSourceCodeWordFinderTest.java

```diff
@@ -34,24 +36,72 @@ public void shouldCheckCamelCaseNameAndReturnThreeErrorTest() throws Exception {
     }
 
     @Test
-    public void shouldCheckMixedNameAndReturnMinusOneThatMeansNoErrorTest() throws Exception {
+    public void shouldCheckMixedNameAndReturnOneErrorTest() throws Exception {
         String testLine = "myWrong111CameeelTest";
         int errorsSize = getErrorsSize(testLine);
-        assertEquals(ERROR_MESSAGE, -1, errorsSize);
+        assertEquals("Wrong error size. Expected = 1", 1, errorsSize);
     }
 
     @Test
     public void shouldCheckMixedNameAndReturnTwoErrorTest() throws Exception {
         String testLine = "myyWrong111CameeelTeest";
         int errorsSize = getErrorsSize(testLine);
-        assertEquals("Wrong error size. Expected = 2", 2, errorsSize);
+        assertEquals("Wrong error size. Expected = 3", 3, errorsSize);
     }
 
     @Test
     public void shouldCheckMixedNameAndReturnTreeErrorTest() throws Exception {
         String testLine = "myyWrong111NaameCameeelTeest";
         int errorsSize = getErrorsSize(testLine);
-        assertEquals("Wrong error size. Expected = 3", 3, errorsSize);
+        assertEquals("Wrong error size. Expected = 4", 4, errorsSize);
+    }
+
+    @Test
+    public void shouldCheckWordWithOneEndDigitAndReturnNoErrorsTest() {
```

Comment:
rewritten with junit theories and data points
https://github.com/webdizz/sonar-grammar/commit/3632ed240151bd00e7c74ed389cd5c7a8f046b51